agent/api/task/task_windows.go

```diff
@@ -213,3 +181,86 @@ func (task *Task) GetCredentialSpecResource() ([]taskresource.TaskResource, bool
 	res, ok := task.ResourcesMapUnsafe[credentialspec.ResourceName]
 	return res, ok
 }
+
+// requiresFSxWindowsFileServerResource returns true if at least one volume in the task
+// is of type 'fsxWindowsFileServer'
+func (task *Task) requiresFSxWindowsFileServerResource() bool {
+	for _, volume := range task.Volumes {
+		if volume.Type == FSxWindowsFileServerVolumeType {
+			return true
+		}
+	}
+	return false
+}
+
+// initializeFSxWindowsFileServerResource builds the resource dependency map for the fsxwindowsfileserver resource
+func (task *Task) initializeFSxWindowsFileServerResource (cfg *config.Config, credentialsManager credentials.Manager,
+	resourceFields *taskresource.ResourceFields) error {
+	for i, vol := range task.Volumes {
+		if vol.Type != FSxWindowsFileServerVolumeType {
+			continue
+		}
+
+		fsxWindowsFileServerVol, ok := vol.Volume.(*fsxwindowsfileserver.FSxWindowsFileServerVolumeConfig)
```

Comment:
nit: Is there a method that can be used to obtain the config here

Reply:
EFS obtains the config in a similar way - https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task.go#L552